sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1883,6 +1883,60 @@ class HiveDDLSuite
     }
   }
 
+  test("SPARK-26836: support Avro schema evolution") {
+    withTable("t") {
+      val originalSchema =
+        """
+          |{
+          |  "namespace": "test",
+          |  "name": "some_schema",
+          |  "type": "record",
+          |  "fields": [
+          |    {
+          |      "name": "col2",
+          |      "type": "string"
+          |    }
+          |  ]
+          |}
+        """.stripMargin
+      sql(
+        s"""
+           |CREATE TABLE t PARTITIONED BY (ds string)
+           |ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe'
+           |WITH SERDEPROPERTIES ('avro.schema.literal'='$originalSchema')
+           |STORED AS
+           |INPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat'
+           |OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat'
+        """.stripMargin)
+      sql("INSERT INTO t partition (ds='1981-01-07') VALUES ('col2_value')")
+      val evolvedSchema =
+        """
+          |{
+          |  "namespace": "test",
+          |  "name": "some_schema",
+          |  "type": "record",
+          |  "fields": [
+          |    {
+          |      "name": "col1",
+          |      "type": "string",
+          |      "default": "col1_default"
+          |    },
+          |    {
+          |      "name": "col2",
+          |      "type": "string"
+          |    }
+          |  ]
+          |}
+        """.stripMargin
+      sql(s"""ALTER TABLE t SET SERDEPROPERTIES ('avro.schema.literal'='$evolvedSchema')""")
+      sql("INSERT INTO t partition (ds='1983-04-27') VALUES ('col1_value', 'col2_value')")
+      withSQLConf(SQLConf.HIVE_AVRO_SCHEMA_EVOLUTION_ENABLED.key -> "true") {
+        checkAnswer(spark.table("t"), Row("col1_default", "col2_value", "1981-01-07")
+          :: Row("col1_value", "col2_value", "1983-04-27") :: Nil)
+      }
```

Comment:
Maybe also `checkAnswer` without the new schema evolution config enabled?

Reply:
I have my doubts regarding this: I mean to validate a bad behavior where one value is given in the wrong column. 
Do we want to freeze the current bad behavior? and why would be that specific wrong result better than all the others?

Reply:
We have the config default to `false` to ensure we preserve the old behavior for compatibility reasons, so it seems to me like we should have a test to verify that we have actually preserved the old behavior. A bug present for long enough eventually starts to become a feature 🤷🏼‍♂️

Basically my take is: if we don't care about compatibility then we shouldn't even have a config, if we do care about compatibility then we should validate it with a test.

But your point is valid so I am open to differing opinions.

Reply:
I mean there are different use cases depending on how the schema is changed. 

Here we have an example where the value in wrong column makes the old behavior practically unusable. 

But for cases when just the default value is changed that sounds at least acceptable: 
all the data is read back with the value which was used during the write (including the missing values).
What about adding such a case for the `SQLConf.HIVE_AVRO_SCHEMA_EVOLUTION_ENABLED.key -> "false"`?

Reply:
Yes, that sounds perfect. I think this is my fault for not reading the test case closely enough and thus making a bad suggestion, sorry about that. Thanks for figuring out what I really wanted :)

Reply:
During the development of our new test I realized to use default value of the schema is only possible via adding a new column. As NULL is different from the missing value thus will not be mapped to default but to the "null" value in the file.
And INSERT always requires to have a value specified  for each of the column.

Based on these new findings I am deplementing `spark.sql.hive.avroSchemaEvolution.enabled` and always use the table schema when available.